### PR TITLE
feat: align fastify.hasRoute to fmw.hasRoute

### PR DESCRIPTION
**Pull request by dancastillo**
#### Checklist

closes: https://github.com/fastify/fastify/issues/5092

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: lib/route.js [hunk 1]
```diff
@@ -148,11 +148,11 @@ function buildRouting (options) {
   }
 
   function hasRoute ({ options }) {
-    return router.find(
+    return router.hasRoute(
       options.method,
       options.url || '',
       options.constraints
-    ) !== null
+    )
```
Comment (gurgunday) [inline lib/route.js:155]: The return type here should stay a boolean

```suggestion
    ) !== null
```

Comment (metcoder95) [reply lib/route.js:155]: Already returns a `boolean` isn't it?
https://github.com/delvedor/find-my-way?tab=readme-ov-file#hasroute-method-path-constraints

File: test/has-route.test.js [hunk 2]
```diff
@@ -71,7 +71,7 @@ test('hasRoute', t => {
 
     t.equal(fastify.hasRoute({
       method: 'GET',
-      url: '/example/12345.png'
+      url: '/example/:file(^\\d+).png'
```
Comment (metcoder95) [inline test/has-route.test.js:74]: Ay any reason for the change?

Comment (Uzlopak) [reply test/has-route.test.js:74]: hasRoute is now a strict matcher. Before it was checking if one route is matching the pattern.

File: lib/route.js (hunk 1 above)
Comment (gurgunday) [reply lib/route.js:155]: Yeah, just woke up 😁🤣

**Review by climba03003 — APPROVED** [review]

**Review by metcoder95 — APPROVED** [review]

**Comment by Uzlopak** [comment]
Marking it as semver-minor

**Comment by mcollina** [comment]
It's currently a major landing in v5.

**Review by mcollina — APPROVED** [review]
lgtm

**Comment by dancastillo** [comment]
@mcollina fixed some tests for the CI process

**Comment by Uzlopak** [comment]
You have to increase the test coverage...

**Comment by dancastillo** [comment]
> You have to increase the test coverage...

hello @Uzlopak can you elaborate, I'm not sure what you mean. The CI check for coverage passes [here](https://github.com/fastify/fastify/actions/runs/6960599936/job/18940462464?pr=5102)

**Comment by Uzlopak** [comment]
See. So some lines are not covered.

image

**Comment by dancastillo** [comment]
Can we rerun failed checks? I do not have permission but [passing](https://github.com/dancastillo/fastify/pull/1) in forked repo
Screenshot 2023-11-22 at 2 08 42 PM

**Comment by Uzlopak** [comment]
Ok, retriggered the failing jobs.

**Comment by Uzlopak** [comment]
The ci will work again when we merge #5193 . Then this PR can be merged too

File: test/findRoute.test.js [hunk 3]
```diff
@@ -49,6 +49,22 @@ test('findRoute should return an immutable route to avoid leaking and runtime ro
   t.same(route.params, { artistId: ':artistId' })
 })
 
+test('findRoute should return null when when url is not passed', t => {
```
Comment (dancastillo) [inline test/findRoute.test.js:52]: this test for or use case of empty string [here](https://github.com/fastify/fastify/pull/5102/files#diff-cf6996cafd1df0bc168a5c5f3a7470f11fc3aea9f908091d5dd6be2e7b30a2c3R166)

**Comment by dancastillo** [comment]
@Uzlopak @mcollina fixed the merge conflicts on here. Should this be good to be merged in?

**Review by mcollina — APPROVED** [review]
lgtm

**Pull request merged**